src/groundlight/internalapi.py

```diff
@@ -70,6 +73,49 @@ class GroundlightApiClient(ApiClient):
     """
 
     REQUEST_ID_HEADER = "X-Request-Id"
+    MAX_CONNECTION_RETRY_ATTEMPTS = 4
+    # Initial delay time of 100 milliseconds
+    RETRY_INITIAL_DELAY = 0.1
+    RETRY_EXPONENTIAL_BACKOFF = 2
+    some_variable = 0
+
+    # Defines the range of HTTP error status codes for which the API
+    # will attempt to re-establish a connection
+    STATUS_CODE_RANGE = range(500, 600)
+
+    @staticmethod
+    def _retry_http_connection(callable):
+        """Tries to re-execute the decorated function in case the execution
+        fails due to a server error (HTTP Error code 500 - 599).
+        Retry attempts are executed while exponentially backing off by a factor
+        of 2 with full jitter (picking a random delay time between 0 and the
+        maximum delay time).
+        :param callable: The function to invoke.
+        """
+
+        @wraps(callable)
+        def wrapper(self, *args, **kwargs):
+            delay = self.RETRY_INITIAL_DELAY
+            retry_count = 0
+            while retry_count <= self.MAX_CONNECTION_RETRY_ATTEMPTS:
+                try:
+                    return callable(self, *args, **kwargs)
+
+                except requests.exceptions.HTTPError as error:
+                    status_code = error.response.status_code
+                    if status_code in self.STATUS_CODE_RANGE:
+                        logger.debug(
+                            f"Current HTTP response status: {status_code}."
+                            f"Remaining retries: {self.MAX_CONNECTION_RETRY_ATTEMPTS - retry_count + 1}"
+                        )
+                        random_delay = random.uniform(0, delay)
+                        time.sleep(random_delay)
```

Comment:
Perfect implementation.  I'd add a short comment saying this is implementing a "full-jitter" strategy.